File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int x = nums[i];
            if(hm.containsKey(target-x)) return new int[]{i, hm.get(target-x)};
            hm.put(x,i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm:dict[int] = dict()
        for i,v in enumerate(nums):
            if target-v in hm:
                return [hm[target-v], i]
            hm[v] = i
        return []